### ISSUE TITLE
Panel Notifications break Pyodide Sites

### ISSUE THREAD

**Issue by michaelweinold** [holoviz/panel#7230]
#### ALL software version info

```
panel==1.4.2
pyodide==0.26.2
```

#### Description of expected behavior and the observed behavior

Using a [notification `pn.state.notifications`](https://panel.holoviz.org/reference/global/Notifications.html) breaks entire Pyodide-built sites.

This is rather inconvenient - and a critical blocker for the Panel WASM dashboard I am building right now.

#### Complete, minimal, self-contained example code that reproduces the issue

In a file `test_app.py`:

```python
import panel as pn

pn.extension(notifications=True)


def show_notification(event):
    pn.state.notifications.error('Sample error message!', duration=60000)


button = pn.widgets.Button(name='Show Notification')
button.on_click(show_notification)

pn.Column(button).servable()
```

Then run, [according to the "converting Panel applications" guide](https://panel.holoviz.org/how_to/wasm/convert.html):

```bash
panel convert dev_app.py --to pyodide-worker --out test
```

<img width="536" alt="Screenshot 2024-09-04 at 13 27 26" src="https://github.com/user-attachments/assets/7a2ccf78-e4f3-4e66-b8ef-0ca3cca9e38c">

Clicking the button raises an error in the browser console (see below).

#### Stack traceback and/or browser JavaScript console output

```
pyodide.asm.js:9 Uncaught (in promise) PythonError: Traceback (most recent call last):
  File "/lib/python311.zip/_pyodide/_base.py", line 573, in eval_code_async
    await CodeRunner(
  File "/lib/python311.zip/_pyodide/_base.py", line 393, in run_async
    coroutine = eval(self.code, globals, locals)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<exec>", line 3, in <module>
  File "/lib/python3.11/site-packages/bokeh/document/document.py", line 391, in apply_json_patch
    DocumentPatchedEvent.handle_event(self, event, setter)
  File "/lib/python3.11/site-packages/bokeh/document/events.py", line 245, in handle_event
    event_cls._handle_event(doc, event)
  File "/lib/python3.11/site-packages/bokeh/document/events.py", line 280, in _handle_event
    cb(event.msg_data)
  File "/lib/python3.11/site-packages/bokeh/document/callbacks.py", line 390, in trigger_event
    model._trigger_event(event)
  File "/lib/python3.11/site-packages/bokeh/util/callback_manager.py", line 113, in _trigger_event
    self.document.callbacks.notify_event(cast(Model, self), event, invoke)
  File "/lib/python3.11/site-packages/bokeh/document/callbacks.py", line 260, in notify_event
    invoke_with_curdoc(doc, callback_invoker)
  File "/lib/python3.11/site-packages/bokeh/document/callbacks.py", line 443, in invoke_with_curdoc
    return f()
           ^^^
  File "/lib/python3.11/site-packages/bokeh/util/callback_manager.py", line 109, in invoke
    cast(EventCallbackWithEvent, callback)(event)
  File "/lib/python3.11/site-packages/panel/reactive.py", line 498, in _server_event
    self._comm_event(doc, event)
  File "/lib/python3.11/site-packages/panel/reactive.py", line 485, in _comm_event
    state._handle_exception(e)
  File "/lib/python3.11/site-packages/panel/io/state.py", line 458, in _handle_exception
    raise exception
  File "/lib/python3.11/site-packages/panel/reactive.py", line 483, in _comm_event
    self._process_bokeh_event(doc, event)
  File "/lib/python3.11/site-packages/panel/reactive.py", line 420, in _process_bokeh_event
    self._process_event(event)
  File "/lib/python3.11/site-packages/panel/widgets/button.py", line 243, in _process_event
    self.clicks += 1
    ^^^^^^^^^^^
  File "/lib/python3.11/site-packages/param/parameterized.py", line 528, in _f
    instance_param.__set__(obj, val)
  File "/lib/python3.11/site-packages/param/parameterized.py", line 530, in _f
    return f(self, obj, val)
           ^^^^^^^^^^^^^^^^^
  File "/lib/python3.11/site-packages/param/parameters.py", line 543, in __set__
    super().__set__(obj,val)
  File "/lib/python3.11/site-packages/param/parameterized.py", line 530, in _f
    return f(self, obj, val)
           ^^^^^^^^^^^^^^^^^
  File "/lib/python3.11/site-packages/param/parameterized.py", line 1553, in __set__
    obj.param._call_watcher(watcher, event)
  File "/lib/python3.11/site-packages/param/parameterized.py", line 2526, in _call_watcher
    self_._execute_watcher(watcher, (event,))
  File "/lib/python3.11/site-packages/param/parameterized.py", line 2506, in _execute_watcher
    watcher.fn(*args, **kwargs)
  File "<exec>", line 16, in show_notification
AttributeError: 'NoneType' object has no attribute 'error'

    at new_error (pyodide.asm.js:9:10014)
    at pyodide.asm.wasm:0x158e62
    at pyodide.asm.wasm:0x15ff9e
    at _PyEM_TrampolineCall_JS (pyodide.asm.js:9:120223)
    at pyodide.asm.wasm:0x1a3494
    at pyodide.asm.wasm:0x28a2bd
    at pyodide.asm.wasm:0x1e431e
    at pyodide.asm.wasm:0x1a397d
    at pyodide.asm.wasm:0x1a3c3e
    at pyodide.asm.wasm:0x1a3ce0
    at pyodide.asm.wasm:0x2689a4
    at pyodide.asm.wasm:0x26e800
    at pyodide.asm.wasm:0x1a3e08
    at pyodide.asm.wasm:0x1a3a98
    at pyodide.asm.wasm:0x15f76b
    at Module.callPyObjectKwargs (pyodide.asm.js:9:64297)
    at Module.callPyObject (pyodide.asm.js:9:65365)
    at wrapper (pyodide.asm.js:9:32855)
```

#### Screenshots or screencasts of the bug in action

- [x] I may be interested in making a pull request to address this

**Comment by michaelweinold** [comment]
Potentially related:

 - https://github.com/holoviz/panel/issues/5090

**Comment by michaelweinold** [comment]
@philippjfr, unfortunately this remains an issue on Panel `1.5.0rc1`. The MWE from above still returns an error of the kind:

```
    watcher.fn(*args, **kwargs)
  File "<exec>", line 17, in show_notification
AttributeError: 'NoneType' object has no attribute 'error'
```

<details>

<summary>JavaScript Console Output</summary>

```
pyodide.asm.js:10 Uncaught (in promise) PythonError: Traceback (most recent call last):
  File "/lib/python312.zip/_pyodide/_base.py", line 596, in eval_code_async
    await CodeRunner(
  File "/lib/python312.zip/_pyodide/_base.py", line 410, in run_async
    coroutine = eval(self.code, globals, locals)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<exec>", line 3, in <module>
  File "/lib/python3.12/site-packages/bokeh/document/document.py", line 391, in apply_json_patch
    DocumentPatchedEvent.handle_event(self, event, setter)
  File "/lib/python3.12/site-packages/bokeh/document/events.py", line 244, in handle_event
    event_cls._handle_event(doc, event)
  File "/lib/python3.12/site-packages/bokeh/document/events.py", line 279, in _handle_event
    cb(event.msg_data)
  File "/lib/python3.12/site-packages/bokeh/document/callbacks.py", line 400, in trigger_event
    model._trigger_event(event)
  File "/lib/python3.12/site-packages/bokeh/util/callback_manager.py", line 111, in _trigger_event
    self.document.callbacks.notify_event(cast(Model, self), event, invoke)
  File "/lib/python3.12/site-packages/bokeh/document/callbacks.py", line 262, in notify_event
    invoke_with_curdoc(doc, callback_invoker)
  File "/lib/python3.12/site-packages/bokeh/document/callbacks.py", line 453, in invoke_with_curdoc
    return f()
           ^^^
  File "/lib/python3.12/site-packages/bokeh/util/callback_manager.py", line 107, in invoke
    cast(EventCallbackWithEvent, callback)(event)
  File "/lib/python3.12/site-packages/panel/reactive.py", line 572, in _server_event
    self._comm_event(doc, event)
  File "/lib/python3.12/site-packages/panel/reactive.py", line 559, in _comm_event
    state._handle_exception(e)
  File "/lib/python3.12/site-packages/panel/io/state.py", line 464, in _handle_exception
    raise exception
  File "/lib/python3.12/site-packages/panel/reactive.py", line 557, in _comm_event
    self._process_bokeh_event(doc, event)
  File "/lib/python3.12/site-packages/panel/reactive.py", line 494, in _process_bokeh_event
    self._process_event(event)
  File "/lib/python3.12/site-packages/panel/widgets/button.py", line 242, in _process_event
    self.clicks += 1
    ^^^^^^^^^^^
  File "/lib/python3.12/site-packages/param/parameterized.py", line 528, in _f
    instance_param.__set__(obj, val)
  File "/lib/python3.12/site-packages/param/parameterized.py", line 530, in _f
    return f(self, obj, val)
           ^^^^^^^^^^^^^^^^^
  File "/lib/python3.12/site-packages/param/parameters.py", line 543, in __set__
    super().__set__(obj,val)
  File "/lib/python3.12/site-packages/param/parameterized.py", line 530, in _f
    return f(self, obj, val)
           ^^^^^^^^^^^^^^^^^
  File "/lib/python3.12/site-packages/param/parameterized.py", line 1553, in __set__
    obj.param._call_watcher(watcher, event)
  File "/lib/python3.12/site-packages/param/parameterized.py", line 2526, in _call_watcher
    self_._execute_watcher(watcher, (event,))
  File "/lib/python3.12/site-packages/param/parameterized.py", line 2506, in _execute_watcher
    watcher.fn(*args, **kwargs)
  File "<exec>", line 17, in show_notification
AttributeError: 'NoneType' object has no attribute 'error'

    at new_error (pyodide.asm.js:10:9965)
    at pyodide.asm.wasm:0x16dc69
    at pyodide.asm.wasm:0x177492
    at _PyEM_TrampolineCall_JS (pyodide.asm.js:10:124082)
    at pyodide.asm.wasm:0x1c2c4c
    at pyodide.asm.wasm:0x2c79ac
    at pyodide.asm.wasm:0x20a621
    at pyodide.asm.wasm:0x1c3339
    at pyodide.asm.wasm:0x1c3648
    at pyodide.asm.wasm:0x1c36c6
    at pyodide.asm.wasm:0x29e6fa
    at pyodide.asm.wasm:0x2a4cf1
    at pyodide.asm.wasm:0x1c3806
    at pyodide.asm.wasm:0x1c346f
    at pyodide.asm.wasm:0x176af6
    at callPyObjectKwargs (pyodide.asm.js:10:62217)
    at Module.callPyObjectMaybePromising (pyodide.asm.js:10:63465)
    at wrapper (pyodide.asm.js:10:27093)
    at Kn.e.port1.onmessage (pyodide.asm.js:10:100342)
```

</details>

<img width="1728" alt="Screenshot 2024-09-08 at 13 03 46" src="https://github.com/user-attachments/assets/ebb8a2f4-5545-46f7-bcb7-dc9ef1ae7f16">

**Comment by philippjfr** [comment]
The fix was only just merged it's not in 1.5.0rc1

**Comment by michaelweinold** [comment]
I would be grateful for another release candidate tag to start testing - installing Panel directly from GitHub through `pip install git+https://github.com/holoviz/panel.git` does not work for me. Thank you!
